kubernetes/client/ws_client_test.py

```diff
@@ -32,6 +36,24 @@ def test_websocket_client(self):
                 ]:
             self.assertEqual(get_websocket_url(url), ws_url)
 
+    @patch.object(websocket._http, '_can_use_sni')
+    @patch.object(websocket._http, '_open_socket')
+    @patch.object(ssl.SSLContext, 'load_verify_locations')
+    def test_use_of_correct_ca_cert(self, load_verify_locations, *ignored):
+        # arrange
+        expected = '/expected-ca.pem'
+        configuration.ssl_ca_cert = expected
+        load_verify_locations.side_effect = Exception("need not go further")
+
+        # act
+        with self.assertRaises(Exception):
```

Comment:
Works until the websocket library elects to change their _private_ implementation and branch to such behavior somewhere else. Point being that stubbing implementation details in external code is always brittle and the best one can do is minimize the number of instructions executed in such a "black box". The chosen option was the least risky I could find, as it minimizes the execution in (by this project) uncontrolled code, to the absolute minimal extent required to assert the side-effect it requires.